src/Files.Uwp/ViewModels/FolderSettingsViewModel.cs

```diff
@@ -1,22 +1,30 @@
 ﻿using Files.Shared.Enums;
 using Files.Uwp.EventArguments;
 using Files.Uwp.Helpers;
+using Files.Uwp.Helpers.LayoutPreferences;
 using Files.Backend.Services.Settings;
 using Files.Uwp.Views.LayoutModes;
 using CommunityToolkit.Mvvm.ComponentModel;
 using CommunityToolkit.Mvvm.DependencyInjection;
 using CommunityToolkit.Mvvm.Input;
-using Newtonsoft.Json;
 using System;
-using System.Linq;
 using System.Windows.Input;
 using Windows.Storage;
+using System.Threading.Tasks;
+using Files.Uwp.Filesystem.StorageItems;
+using Files.Uwp.Filesystem;
+using Files.Shared.Extensions;
+using Newtonsoft.Json;
+using IO = System.IO;
 
 namespace Files.Uwp.ViewModels
 {
     public class FolderSettingsViewModel : ObservableObject
     {
-        private static readonly ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
+        public static string LayoutSettingsDbPath => IO.Path.Combine(ApplicationData.Current.LocalFolder.Path, "layoutsettings.db");
+
+        private static readonly Lazy<LayoutPrefsDb> dbInstance = new(() => new LayoutPrefsDb(LayoutSettingsDbPath, true));
+        public static LayoutPrefsDb DbInstance => dbInstance.Value;
```

Comment:
Out of curiosity, what would happen if for some reason DB creation takes ages or fails?

Reply:
I've looked at the source code for LiteDb and it does not do anything much in the constructor (not even creating the DB file). So this Lazy<> initialization is probably redundant.